web/src/features/AppConfig/components/AppConfig.tsx

```diff
@@ -427,6 +435,26 @@ class AppConfig extends Component<Props, State> {
     return foundItem;
   };
 
+  validateRegex = ({
+    validation,
+    value,
+  }: {
+    validation: Validation;
+    value: string;
+  }): Validation => {
+    if (!value || !validation?.regex) {
+      return { ...validation, hasError: false };
+    }
+
+    const validationRegex = new RegExp(validation.regex);
+
+    if (validationRegex.test(value)) {
+      return { ...validation, hasError: false };
+    } else {
+      return { ...validation, hasError: true };
+    }
+  };
```

Comment:
This is kind of clunky but it takes a Validation and a value and returns a Validation. This function primarily modifies `hasErrors`